extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/datalink/MongoDataLink.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.jet.mongodb.datalink;
+
+import com.hazelcast.config.DataLinkConfig;
+import com.hazelcast.datalink.DataLinkBase;
+import com.hazelcast.datalink.DataLinkResource;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.hazelcast.internal.util.Preconditions.checkNotNull;
+import static com.hazelcast.internal.util.Preconditions.checkState;
+
+/**
+ * Creates a MongoDB DataLink.
+ *
+ * According to {@link MongoClient} documentation, the client object is responsible for maintaining connection pool,
+ * so this data link just returns one, cached client.
+ *
+ * @since 5.3
+ */
+public class MongoDataLink extends DataLinkBase {
+    /**
+     * Name of a property which holds connection string to the mongodb instance.
+     */
+    public static final String CONNECTION_STRING_PROPERTY = "connectionString";
+    private MongoClient mongoClient;
+
+    /**
+     * Creates a new data link based on given config.
+     */
+    public MongoDataLink(DataLinkConfig config) {
+        super(config);
+        this.mongoClient = new CloseableMongoClient(createClient(config), this::release);
```

Comment:
you can guard this by `if (getConfig().isShared())` so we don't create an unnecessary instance.